src/main/kotlin/com/ampnet/reportservice/service/impl/XlsxServiceImpl.kt

```diff
@@ -0,0 +1,144 @@
+                    .filter { depositOwners.contains(it.uuid) }
+                mapUserResponse(usersWithDeposit)
+            }
+            XlsxType.INVESTMENT -> {
+                // users with at least one investment
+                val activeUsers = userService.getAllActiveUsers(coop).usersList
+                val walletsWithInvestment = blockchainService.getUserWalletsWithInvestment(coop).map { it.wallet }
+                val walletOwners = walletService
+                    .getWalletsByOwner(activeUsers.map { UUID.fromString(it.uuid) })
+                    .filter { walletsWithInvestment.contains(it.activationData) }
+                    .map { it.owner }
+                val usersWithInvestment = activeUsers.filter { walletOwners.contains(it.uuid) }
+                mapUserResponse(usersWithInvestment)
+            }
+        }
+
+    private fun writeHeaderLine(type: XlsxType) {
+        sheet = workbook.createSheet("Users-$type")
+        val font = workbook.createFont().apply {
+            bold = true
+            fontHeight = headerFontSize
+        }
+        val style = workbook.createCellStyle().apply { setFont(font) }
+        val row = sheet.createRow(0)
+        var columnCount = 0
+        createCell(row, columnCount++, "User UUID", style)
+        createCell(row, columnCount++, "E-mail", style)
+        createCell(row, columnCount++, "First Name", style)
+        createCell(row, columnCount++, "Last Name", style)
+        createCell(row, columnCount++, "Authentication Method", style)
+        createCell(row, columnCount, "Registration Date", style)
+    }
+
+    private fun writeDataLines(users: List<UserResponse>) {
+        val font = workbook.createFont().apply { fontHeight = dataFontSize }
+        val style = workbook.createCellStyle().apply { setFont(font) }
+        var rowCount = 1
+        for (user in users) {
+            val row: Row = sheet.createRow(rowCount++)
+            var columnCount = 0
+            createCell(row, columnCount++, user.uuid, style)
+            createCell(row, columnCount++, user.email, style)
+            createCell(row, columnCount++, user.firstName, style)
+            createCell(row, columnCount++, user.lastName, style)
+            createCell(row, columnCount++, user.auth, style)
+            createCell(row, columnCount, dateToString(user.createdAt), style)
+        }
+    }
+
+    private fun createCell(row: Row, columnCount: Int, value: String, style: CellStyle) {
+        sheet.autoSizeColumn(columnCount)
+        row.createCell(columnCount).apply {
+            setCellValue(value)
+            cellStyle = style
+        }
+    }
+
+    private fun dateToString(date: Long): String =
+        date.millisecondsToLocalDateTime().format(DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss"))
+
+    private fun mapUserResponse(users: List<UserExtendedResponse>): List<UserResponse> =
+        users.map {
+            UserResponse.newBuilder()
```

Comment:
Can't you use `apply` for this? Just change `users.map {` to `users.map { user ->` and then:
```kotlin
UserResponse.newBuilder().apply {
    uuid = user.uuid
    ...
}.build()
```